### ISSUE TITLE
Creating openstack_lb_listener_v2 fails with Contrail Neutron plugin in v.1.14 and greather

### ISSUE THREAD

**Issue by MirgDenis** [terraform-providers/terraform-provider-openstack#687]
Hi,

I have issue with Contrail Neutron plugin and Terraform Openstack provider version 1.14 and greather. In such configuration I can`t add listeners to loadbalancer. This probably happens because function statuses is not defined in Contrail Neutron plugin:
```
    def stats(self, context, loadbalancer_id):
        pass

    def statuses(self, context, loadbalancer_id):
        pass
```
But in previous version of Terraform Openstack provider this resource was working.

### Terraform Version

```
$ terraform -v
Terraform v0.11.11
+ provider.openstack v1.14.0
```

### Affected Resource(s)
- openstack_lb_listener_v2

### Terraform Configuration Files
```
provider "openstack" {
  version = "= 1.14"
}

resource "openstack_networking_network_v2" "network_internal" {
  name                = "network-internal"
  admin_state_up      = "true"
}

resource "openstack_networking_subnet_v2" "subnet_internal" {
  name                = "subnet-internal"
  network_id          = "${openstack_networking_network_v2.network_internal.id}"
  cidr                = "10.0.0.0/8"
  ip_version          = 4
}

resource "openstack_lb_loadbalancer_v2" "lb_1" {
 vip_subnet_id      = "${openstack_networking_subnet_v2.subnet_internal.id}"
 vip_address        = "10.0.0.10"
}

## This resource fails
resource "openstack_lb_listener_v2" "listener_1" {
 protocol        = "HTTP"
 protocol_port   = 80
 name            = "lala"
 loadbalancer_id = "${openstack_lb_loadbalancer_v2.lb_1.id}"
}
```

### Debug Output
```
2019-02-22T16:31:11.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:11 [DEBUG] Waiting for listener de6ff30b-4db6-49ea-ba37-ad77d6ee740d to become ACTIVE.
2019-02-22T16:31:11.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:11 [DEBUG] Waiting for state to become: [ACTIVE]
2019-02-22T16:31:12.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:12 [DEBUG] OpenStack Request URL: GET http://10.10.1.10:9696/v2.0/lbaas/loadbalancers/9effe6e9-c6c0-4f6b-b4c3-07998b6437cd/statuses
2019-02-22T16:31:12.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:12 [DEBUG] OpenStack Request Headers:
2019-02-22T16:31:12.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: Accept: application/json
2019-02-22T16:31:12.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: User-Agent: Terraform/0.11.7 gophercloud/2.0.0
2019-02-22T16:31:12.276+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: X-Auth-Token: ***
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:12 [DEBUG] OpenStack Response Code: 200
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:12 [DEBUG] OpenStack Response Headers:
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: Content-Length: 4
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: Content-Type: application/json
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: Date: Fri, 22 Feb 2019 14:31:12 GMT
2019-02-22T16:31:12.532+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: X-Openstack-Request-Id: req-a01e5528-eef2-49ac-a145-f93dd7a22a74
2019-02-22T16:31:12.533+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 2019/02/22 16:31:12 [DEBUG] OpenStack Response Body: null
2019-02-22T16:31:12.535+0200 [DEBUG] plugin.terraform-provider-openstack_v1.14.0_x4: 
Error: Error applying plan:

1 error(s) occurred:

* openstack_lb_listener_v2.listener_1: 1 error(s) occurred:

* openstack_lb_listener_v2.listener_1: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Actual Behavior
`terraform apply` failed but listener is added
```root@ctl01:~# neutron lbaas-loadbalancer-show 58eb7b2c-c08e-4b44-a591-026fd7fb94d4
neutron CLI is deprecated and will be removed in the future. Use openstack CLI instead.
+---------------------+------------------------------------------------+
| Field               | Value                                          |
+---------------------+------------------------------------------------+
| admin_state_up      | True                                           |
| description         |                                                |
| id                  | de6ff30b-4db6-49ea-ba37-ad77d6ee740d           |
| listeners           | {"id": "9effe6e9-c6c0-4f6b-b4c3-07998b6437cd"} |
| name                |                                                |
| operating_status    | ONLINE                                         |
| provider            | opencontrail                                   |
| provisioning_status | ACTIVE                                         |
| tenant_id           | a50f8bdbc0e34db8bec1141309077b9f               |
| vip_address         | 10.0.0.10                                      |
| vip_port_id         | 4739dbe2-b09f-4956-ac8e-0bf550ca2a25           |
| vip_subnet_id       | c4180763-5715-497e-9af7-f36be49e631f           |
+---------------------+------------------------------------------------+
```

**Comment by jtopjian** [comment]
@MirgDenis Thank you for reporting this. Would you be able to confirm if the fix in #691 resolves this issue?

**Comment by MirgDenis** [comment]
Yes, thanks it helped.
